File: PathSum/I/PathSum.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    def hasPathSum(self, root, sum):
        """
        :type root: TreeNode
        :type sum: int
        :rtype: bool
        """
        if root is None:
        	return False
        elif root.left is None and root.right is None and root.val == sum:
        	return True
        else:
        	left = self.hasPathSum(root.left, sum - root.val)
        	right = self.hasPathSum(root.right, sum - root.val)
        	return left | right

File: PathSum/I/PathSum.cpp
/**
* Definition for a binary tree node.
*/
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};

class Solution {
public:
	bool hasPathSum(TreeNode* root, int sum) {
		if (root == nullptr)
			return false;
		if (root->left == nullptr && root->right == nullptr && root->val == sum)
			return true;
		else {
			bool left =  hasPathSum(root->left, sum - root->val);
			bool right =  hasPathSum(root->right, sum - root->val);
			return left | right;
		}
	}
};